src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp

```diff
@@ -616,22 +618,58 @@ void Transformations::MainSnippets(void) {
         !dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx2)) // snippets are implemented only for relevant platforms (avx2+ extensions)
         return;
 
+    // At the moment Snippets supports Transposes in MHA pattern only in FP32 case since
+    //      - ConvertSaturation[BF16->FP32] will be inserted after Parameters and before Transposes in canonicalization stage
+    //      - ConvertSaturation[FP32->BF16] will be inserted after Transposes and before Brgemm in precision propagation stage
+    // Because of that Transposes won't be fused into Brgemm
+    // TODO [111813]: Need to update this pipeline to avoid Converts between Transposes and Brgemm on inputs
+    ov::snippets::pass::SnippetsTokenization::Config tokenization_config;
+    tokenization_config.mha_token_enable_transpose = !enableBF16;
+
     ngraph::pass::Manager snippetsManager;
     snippetsManager.set_per_pass_validation(false);
     if (snippetsMode != Config::SnippetsMode::IgnoreCallback)
         CPU_REGISTER_PASS_X64(snippetsManager, SnippetsMarkSkipped, enableBF16);
-    CPU_REGISTER_PASS_X64(snippetsManager, snippets::pass::SnippetsTokenization);
+    CPU_REGISTER_PASS_X64(snippetsManager, snippets::pass::SnippetsTokenization, tokenization_config);
 
+    // Tokenize MHA in quantized model or with BF16 only in tests.
+    // TODO [106921]: Please enable the tokenization when the ticket 106921 with blocking support for BRGEMM will be implemented
+    const bool onlyFloatSupported = snippetsMode != Config::SnippetsMode::IgnoreCallback;
     const bool isMHASupported =
-            !enableBF16 &&  // TODO: Need to add BF16 support for MHA in Snippets
+            IMPLICATION(enableBF16, !onlyFloatSupported) &&
             dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core);  // MHA has BRGEMM that is supported only on AVX512 platforms
     if (!isMHASupported) {
         CPU_DISABLE_PASS_X64(snippetsManager, snippets::pass::TokenizeMHASnippets);
     }
+
+#if defined(OPENVINO_ARCH_X86_64)
+    auto is_supported_matmul = [onlyFloatSupported](const std::shared_ptr<const ov::Node>& n) {
+        const auto matmul = ov::as_type_ptr<const ov::op::v0::MatMul>(n);
+        if (!matmul)
+            return false;
+        if (matmul->get_input_element_type(1) == ov::element::i8)
+            return !onlyFloatSupported && dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core_vnni);
+        if (matmul->get_input_element_type(0) == ov::element::bf16 &&
```

Comment:
Is it really required to check bf16 here? In general flow it should appear later in the pipeline. 
Moreover I would expect general precision propogation logic inside Snippets should upscale precsion to FP32 based on CPUTargetMachine feedback (which returns only FP32 for MatMul on based on ISA).

Reply:
Discussed offline. Seems like the `PrecisionPropagation` pass should be called before plugin specific transformation `BrgemmToBrgemmCPU`. But at the same time `PrecisionPropagation` matches on emitters which are unavailable before `BrgemmToBrgemmCPU` call. The is some conflict. The ticket: 112812

Thanks a lot for the idea!